src/styles/utils/loop.scss

```diff
@@ -0,0 +1,20 @@
+@each $day, $dayN in $days-day {
+  .calendar--start-day-#{$day} 
+  .calendar__days:first-child {
+    margin-left: $dayN;
```

Comment:
It should indicate margin for the first day of month depends what is start day (mon tue wed...) and move your first day of calendar, but margins are 0, 1, 2... Try to correct $days-day so it will work correct. Like this - if it's tuesday margin-left should be size of cell (100px) + gap (1px), wednesday - margin-left should be 2 * (size of cell + gap)    etc.